new-toast-site/fetch-mdx-post-files.js

```diff
@@ -1,41 +1,43 @@
-const fs = require("fs").promises;
-const fsReg = require("fs");
-const slugify = require("@sindresorhus/slugify");
-const mdx = require("@mdx-js/mdx");
-const util = require("util");
-const vm = require("vm");
-const rehypePrism = require("./rehype-prism-mdx");
-const rehypeSlug = require("rehype-slug");
-const rehypeLink = require("rehype-autolink-headings");
-const parse = require("rehype-parse");
-const unified = require("unified");
-const globby = require("globby");
-const chalk = require("chalk");
+import { promises as fs } from "fs";
+import slugify from "@sindresorhus/slugify";
+import mdx from "@mdx-js/mdx";
+// import rehypePrism from "./rehype-prism-mdx.js";
+import rehypeSlug from "rehype-slug";
+import rehypeLink from "rehype-autolink-headings";
+import json5 from "json5";
+import parse from "rehype-parse";
+import unified from "unified";
+import visit from "unist-util-visit";
+import globby from "globby";
+import chalk from "chalk";
 
-const {
-  transformComponentForBrowser,
-  transformComponentForNode,
-} = require("toast/src/transforms");
-
-const parseSvg = unified().use(parse, {
-  emitParseErrors: true,
-  duplicateAttribute: false,
-});
-
-exports.sourceData = async ({ createPage, ...options }) => {
+export const sourceData = async ({ createPage, ...options }) => {
   const files = await globby("./content/posts", {
     expandDirectories: { extensions: ["mdx"] },
   });
 
   return Promise.all(
     files.map(async (filename) => {
+      let meta = {};
+      const remarkPluckMeta = (_options) => (tree) => {
```

Comment:
this replaces the vm.run from later in the file, resulting in a faster MDX setup and less compilation work (we don't have to compile and run the code to get the meta json)